src/components/hooks/useFileSelect.ts

```diff
@@ -2,44 +2,51 @@ import { Buffer } from "buffer";
 
 import { useCallback, useState } from "react";
 
-import { FileSelectProps } from "../FileSelect";
+import { FileSelectProps, SelectedFile } from "../FileSelect";
 import { compareMime } from "../helpers";
 
 export default function useFileSelect(props: FileSelectProps) {
     const [dragging, setDragging] = useState<boolean>(false);
 
     const onChange = useCallback(
         async (files: File[]) => {
+            // Filter files by the allowed mimetypes
             if (props.allowedMimeTypes)
                 files = files.filter(
                     (x) => props.allowedMimeTypes.findIndex((y) => compareMime(y, x.type)) !== -1
                 );
 
+            // If no files made it through checks then
+            // return and add no files to the select
             if (!files.length) return;
 
+            // Create a "hashmap" of the current values
```

Comment:
```suggestion
            // Create a hash set of the current values
```
A hashmap is a mapping of `hash -> value`, kinda like a dict where the key is a hash of the value.
What we have here is a hash set

see: https://en.wikipedia.org/wiki/Hash_table